### PR TITLE
CoreRT to NativeAOT renames

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
This is the remaining cleanup for https://github.com/dotnet/runtime/pull/68151#issuecomment-1101686845 . Majority of remaining instances are links to archived issues that does not make sense to cleanup.

File: src/mono/System.Private.CoreLib/src/System/Threading/Overlapped.cs [hunk 1]
```diff
@@ -84,8 +84,7 @@ internal sealed unsafe class OverlappedData
                     }
                 }
 
-                //CORERT: NativeOverlapped* pNativeOverlapped = (NativeOverlapped*)Interop.MemAlloc((UIntPtr)(sizeof(NativeOverlapped) + sizeof(GCHandle)));
-                NativeOverlapped* pNativeOverlapped = (NativeOverlapped*)Marshal.AllocHGlobal((IntPtr)(sizeof(NativeOverlapped) + sizeof(GCHandle)));
+                NativeOverlapped* pNativeOverlapped = (NativeOverlapped*)NativeMemory.Alloc((nuint)(sizeof(NativeOverlapped) + sizeof(GCHandle)));
```
Comment (jkotas) [inline src/mono/System.Private.CoreLib/src/System/Threading/Overlapped.cs:87]: I have converted several other places that used `Marshal.AllocHGlobal` to `NativeMemory.Alloc` while I was on it.

File: src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/AllocFreeConcurrentStack.cs [hunk 2]
```diff
@@ -57,19 +77,15 @@ private static Stack<RefAsValueType<T>> ThreadLocalStack
...
+#if !NETCOREAPP
     internal static class AllocFreeConcurrentStack
     {
-        // WARNING: We allow diagnostic tools to directly inspect this member (t_stacks).
-        // See https://github.com/dotnet/corert/blob/master/Documentation/design-docs/diagnostics/diagnostics-tools-contract.md for more details.
-        // Please do not change the type, the name, or the semantic usage of this member without understanding the implication for tools.
-        // Get in touch with the diagnostics team if you have questions.
-
         // Workaround for https://github.com/dotnet/runtime/issues/4731.
-        // When that's fixed, a [ThreadStatic] Stack should be added back to AllocFreeConcurrentStack<T>.
```
Comment (jkotas) [inline src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/AllocFreeConcurrentStack.cs:70]: This bug has been fixed a long time ago. I have kept the workaround for netstandard builds.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Exception.NativeAot.cs [hunk 3]
```diff
@@ -30,10 +30,6 @@ public partial class Exception
 
         private static string? SerializationWatsonBuckets => null;
 
-        // WARNING: We allow diagnostic tools to directly inspect these three members (_message, _innerException and _HResult)
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Exception.NativeAot.cs:33]: Delete these comments from everywhere. They were specific to .NET Native for UWP debugger that we do not care about anymore.

File: src/coreclr/nativeaot/Runtime/ICodeManager.h [hunk 4]
```diff
@@ -145,8 +145,6 @@ class ICodeManager
                                             PTR_PTR_VOID *  ppvRetAddrLocation,     // out
                                             GCRefKind *     pRetValueKind) = 0;     // out
 
-    virtual void UnsynchronizedHijackMethodLoops(MethodInfo * pMethodInfo) = 0;
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/ICodeManager.h:148]: This was specific to .NET Native for UWP. We do not need it anymore.

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets [hunk 5]
```diff
@@ -13,20 +13,13 @@
     a target called ComputeFilesToPublish which produces the ItemGroup
     ResolvedFileToPublish based on the inputs of @(IntermediateAssembly)
     and @(ResolvedAssembliesToPublish). We modify those two item groups
-    to control what gets published after CoreRT optimizes the application.
-
-    For .NET Core 3.0:
```
Comment (jkotas) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets:18]: Deleting support for older .NET SDKs. We support .NET 6+ SDK only.

File: src/coreclr/inc/cordebuginfo.h [hunk 6]
```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 //
-// Keep in sync with https://github.com/dotnet/corert/blob/master/src/Native/ObjWriter/cordebuginfo.h
+// Keep in sync with https://github.com/dotnet/llvm-project/blob/objwriter/<branch>/llvm/tools/objwriter/cordebuginfo.h
```
Comment (LakshanF) [inline src/coreclr/inc/cordebuginfo.h:5]: Its likely my lack of familiarity with the` llvm-project` repo but I found it a little challenging to find the `coredebuginfo.h` file there

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets [hunk 7]
```diff
@@ -32,18 +32,18 @@
...
-        
+
   </PropertyGroup>
 
   <!-- Locate the runtime package according to the current target runtime -->
   <Target Name="ImportRuntimeIlcPackageTarget" Condition="'$(BuildingFrameworkLibrary)' != 'true' AND $(IlcCalledViaPackage) == 'true'" DependsOnTargets="$(ImportRuntimeIlcPackageTargetDependsOn)" BeforeTargets="Publish">
     <!-- This targets file is imported by the SDK when the PublishAot property is set. SDK resolves runtime package paths differently-->
-    <Error Condition="'@(ResolvedILCompilerPack)' == '' AND '$(PublishAot)' == 'true'" Text="The PackageDefinitions ItemGroup is required for target ImportRuntimeIlcPackageTarget" />
-    <!-- CoreRT SDK and Framework Assemblies need to be defined to avoid CoreCLR implementations being set as compiler inputs -->
+    <Error Condition="'@(ResolvedILCompilerPack)' == '' AND '$(PublishAot)' == 'true'" Text="The ResolvedILCompilerPack ItemGroup is required for target ImportRuntimeIlcPackageTarget" />
```
Comment (LakshanF) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets:45]: Thanks!

**Review by LakshanF — APPROVED** [review]

File: src/coreclr/inc/cordebuginfo.h (hunk 6 above)
Comment (jkotas) [inline src/coreclr/inc/cordebuginfo.h:5]:
```suggestion
// Keep in sync with llvm/tools/objwriter/cordebuginfo.h in current objwriter branch in https://github.com/dotnet/llvm-project repo
```

Comment (jkotas) [reply src/coreclr/inc/cordebuginfo.h]: I have updated the comment that should make navigating to the right file easier.

**Pull request merged**